.gitignore

```
@@ -12,6 +11,7 @@
 *.user
 *.userosscache
 *.sln.docstates
+**/*/appsettings*.json
```

this wasn't included in the original .gitignore, but I assume it shouldn't be committed since the environment variable is set here?

---

I can't see the settings file in question, as it isn't committed, but looking at the use of `<UserSecretsId>d3490a8f-e855-472a-ad82-2b5f4ca9d5a9</UserSecretsId>` in `BusBoard.ConsoleApp.csproj`, and looking at [these docs](https://learn.microsoft.com/en-us/aspnet/core/security/app-secrets?view=aspnetcore-9.0&tabs=windows), are you using user secrets that are stored outside the repo?

Please do let me know if I've misunderstood
 
It would be great to add some details to the readme file to let a new dev know how to set things up to use their TFL api key, e.g. which commands they need to run.

---

I have my secret defined in `%APPDATA%\Microsoft\UserSecrets\<user_secrets_id>\secrets.json` and it defined in `AssemblyInfo.cs` as expected.

Update: FIXED 😁
I was missing `.AddUserSecrets<Program>()` in `Program.cs`:
```
var config = new ConfigurationBuilder()
			.AddUserSecrets<Program>()
			.AddEnvironmentVariables()
			.Build();
```

The secret is being loaded into the program correctly now. I'll add a not in the `readme.md` on how to add your secret to the secret manager.

I'll also add a check for the key in the code as this can fail silently without anyone knowing.

---

Added the missing key check 👍